sf-auth-client/src/main/java/com/ascendix/salesforce/oauth/ForceOAuthClient.java

```diff
@@ -85,31 +101,46 @@ private static void extractPartnerUrl(ForceUserInfo userInfo) {
         userInfo.setPartnerUrl(userInfo.getUrls().get("partner").replace("{version}", API_VERSION));
     }
 
-    private static boolean isBadTokenError(HttpResponseException e) {
-        return e.getStatusCode() == HttpStatusCodes.STATUS_CODE_FORBIDDEN;
+    private boolean isBadTokenError(HttpResponseException e) {
+        return ((e.getStatusCode() == HttpStatusCodes.STATUS_CODE_FORBIDDEN)
+                &&
+                StringUtils.equalsAnyIgnoreCase(responseContent, BAD_TOKEN_SF_ERROR_CODE) ||
```

Comment:
Please use StringUtils.equalsAnyIgnoreCase() in correct way StringUtils.equalsAnyIgnoreCase(responseContent,  BAD_TOKEN_SF_ERROR_CODE,  MISSING_TOKEN_SF_ERROR_CODE, WRONG_ORG_SF_ERROR_CODE)

Reply:
fixed